components/reference-data/src/main/java/org/datacleaner/beans/transform/RemoveDictionaryMatchesTransformer.java

```diff
@@ -98,27 +114,61 @@ public void close() {
     @Override
     public Object[] transform(InputRow inputRow) {
         final String value = inputRow.getValue(_column);
-        final String result = transform(value);
-        return new Object[] { result };
+        final Object[] result = transform(value);
+        return result;
     }
 
-    public String transform(final String value) {
-        if (Strings.isNullOrEmpty(value)) {
-            return value;
+    public Object[] transform(final String value) {
+        if (!_removedMatchesAsList) {
+            return getRemovedMatchesAsString(value);
+        } else {
+            return getRemovedMatchesAsList(value);
         }
+    }
+
+    public String[] getRemovedMatchesAsString(final String value) {
```

Comment:
I don't think the name of this method is correct, since it does the complete transformation not just the latter part.